### PR TITLE
Shelter updates

### PR DESCRIPTION
**Pull request by janegreene**
story 29 still wip needs dynamic flash messages but model is tested and working
pushing wip so we can deploy heroku for show and tell

### CODE REVIEW

File: app/controllers/shelters_controller.rb [hunk 1]
```diff
@@ -11,8 +11,13 @@ def new
   end
 
   def create
-    Shelter.create(shelter_params)
-    redirect_to '/shelters'
+    shelter = Shelter.new(shelter_params)
+    if shelter.save
+        redirect_to '/shelters'
+    else
+        flash.now[:notice] = "Sheleter not created. Required information missing: Address"
```
Comment (janegreene) [inline app/controllers/shelters_controller.rb:18]: hard coded message needs update

File: app/models/shelter.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Shelter < ApplicationRecord
-  validates_presence_of :name
-  has_many :pets, dependent: :delete_all
-  has_many :reviews, dependent: :delete_all
+  validates_presence_of :name, :address, :city, :state, :zip
```
Comment (janegreene) [inline app/models/shelter.rb:2]: updated required fields and dependencies

File: spec/models/shelter_spec.rb [hunk 3]
```diff
@@ -3,11 +3,14 @@
...
+    it {should validate_presence_of :state}
+    it {should validate_presence_of :zip}
   end
-  
-  describe 'relationships' do
-    it {should have_many :pets}
-    it {should have_many :reviews}
 
+  describe 'relationships' do
+    it {should have_many(:reviews).dependent(:destroy) }
```
Comment (whitneykidd) [inline spec/models/shelter_spec.rb:13]: Good job on getting these tests to pass

File: spec/features/shelters/index_spec.rb [hunk 4]
```diff
@@ -65,4 +65,25 @@
...
+  fill_in "City", with: "Cincinnati"
+  fill_in "State", with: "OH"
+  fill_in "Zip", with: 41073
+
+  click_button "Create Shelter"
+  expect(page).to have_content("Sheleter not created. Required information missing: Address")#need to make dynamic
 end
+end
+
+# User Story 29, Flash Messages for Shelter Create and Update
```
Comment (janegreene) [inline spec/features/shelters/index_spec.rb:84]: will delete comments prior to final submission

**Pull request merged**